app/src/conferences/models.py

```diff
@@ -286,41 +286,34 @@ class Meta:
     def _pays_for(self, option_name):
         return getattr(self, option_name)
 
-    def _price_for(self, option_name):
-        breakfast_price = self.zosia.price_accomodation_breakfast - self.zosia.price_accomodation
-        dinner_price = self.zosia.price_accomodation_dinner - self.zosia.price_accomodation
-        return {
-            'accomodation_day_1': self.zosia.price_accomodation,
-            'dinner_1': dinner_price,
-            'breakfast_2': breakfast_price,
-            'accomodation_day_2': self.zosia.price_accomodation,
-            'dinner_2': dinner_price,
-            'breakfast_3': breakfast_price,
-            'accomodation_day_3': self.zosia.price_accomodation,
-            'dinner_3': dinner_price,
-            'breakfast_4': breakfast_price
-        }[option_name]
+    def _price_for(self, chosen):
+        if not chosen["accomodation_day"] and not chosen["dinner"] and not chosen["breakfast"]:
+            return 0
+
+        if chosen["dinner"] and chosen["breakfast"]:
+            return self.zosia.price_whole_day
+
+        if chosen["dinner"] and not chosen["breakfast"]:
+            return self.zosia.price_accomodation_dinner
+
+        if not chosen["dinner"] and chosen["breakfast"]:
+            return self.zosia.price_accomodation_breakfast
+
+        return self.zosia.price_accomodation
 
     @property
     def price(self):
         payment = self.zosia.price_base
 
-        payment_groups = [
-            ['accomodation_day_1', 'dinner_1', 'breakfast_2'],
-            ['accomodation_day_2', 'dinner_2', 'breakfast_3'],
-            ['accomodation_day_3', 'dinner_3', 'breakfast_4'],
-        ]
-
         if self.bus is not None:
             payment += self.zosia.price_transport
 
-        for group in payment_groups:
-            if all(map(lambda d: self._pays_for(d), group)):
-                payment += self.zosia.price_whole_day
-            else:
-                for g in group:
-                    if self._pays_for(g):
-                        payment += self._price_for(g)
+        for accommodation, meals in PAYMENT_GROUPS.items():
+            chosen = {
+                accommodation[:-2]: self._pays_for(accommodation),
```

Comment:
I have to read whole PR to understand why you are removing last 2 characters, please add some comment about it